misk-api/build.gradle.kts

```diff
@@ -19,3 +20,9 @@ configure<MavenPublishBaseExtension> {
     KotlinJvm(javadocJar = Dokka("dokkaGfm"))
   )
 }
+
+tasks.getting(KotlinCompile::class) {
+  compilerOptions {
+    freeCompilerArgs.add("-Xjvm-default=all-compatibility")
```

Comment:
This tells the compiler to make the change binary compatible
https://kotlinlang.org/docs/java-to-kotlin-interop.html#all-compatibility